src/createAnimatedComponent.tsx

```diff
@@ -412,32 +422,51 @@ export default function createAnimatedComponent(
         // update UI props whitelist for this view
         if (
           hostInstance &&
-          this._hasReanimated2Props(styles) &&
+          (this.props.animatedProps?.viewDescriptors || styles.length) &&
```

Comment:
I'd extract this to a variable defined before the if statement to make it clear what this part of the condition means:

```
const hasReanimated2Props = this.props.animatedProps?.viewDescriptors || styles.length;

if (hasReanimated2Props && hostInstance?.viewConfig) {
```

Reply:
Done in f6b8560835b25f1b3463a3bce66b2265060bc3ac